### PR TITLE
TE-1554 / Changing the NFS shares test to the new test format

### PR DESCRIPTION
**Pull request by ericbsd**
Added allure tags, epics, features, and stories.
Added test description and updated common_shares.py docstring.
Added missing example in nfs keyword docstring
image

### CODE REVIEW

File: test_cases/webui/shares/nfs/conftest.py [hunk 1]
```diff
@@ -6,16 +6,11 @@
...
     Common.create_non_admin_user_by_api('nfs_test_user', 'NFS Test User', 'testing', 'True')
-    print('in nfs setup after')
 
 
 @pytest.fixture(scope='class', autouse=True)
 def teardown_class():
     yield
-    print('in nfs finish before')
     COMSHARE.delete_all_shares_by_sharetype('nfs')
     Common.delete_user_by_api('nfs_test_user')
```
Comment (markbainix) [inline test_cases/webui/shares/nfs/conftest.py:16]: we should change these `Common.delete_user_by_api` to `API_DELETE.delete_user` to separate api and webui.

File: test_cases/webui/shares/nfs/test_create_new_nfs_share.py [hunk 2]
```diff
@@ -6,39 +7,59 @@
 from keywords.webui.navigation import Navigation as NAV
 
 
+@allure.tag("NFS Shares", "Create")
```
Comment (markbainix) [inline test_cases/webui/shares/nfs/test_create_new_nfs_share.py:10]: I would put the "Create" tag on the test case only, and leave the more global "NFS  Shares" at the class level. reason is if you add test cases to this class they may not be "create"

File: test_cases/webui/shares/nfs/test_create_new_nfs_share.py [hunk 3]
```diff
@@ -6,39 +7,59 @@
...
+    """
+    This test class covers the NFS share create test cases.
+    """
+
+    @pytest.fixture(scope='function', autouse=True)
+    def setup_test(self, nfs_data):
+        """
+        This fixture sets the dataset for the NFS share test
+        """
+        DATASET.create_dataset_by_api(nfs_data['api_path'], 'NFS')
```
Comment (markbainix) [inline test_cases/webui/shares/nfs/test_create_new_nfs_share.py:24]: need to at least add `COMSHARE.delete_share_by_api('nfs', nfs_data['api_path'])` if share is created elsewhere and left available, the "Create" later will fail with error stating share already exists.

Also suggest moving the Navigation out of test and into setup. This is how we have been doing it, and reduces the test to just performing the "create" actions/step and not system setup steps.

File: test_cases/webui/shares/nfs/test_create_new_nfs_share.py [hunk 4]
```diff
@@ -6,39 +7,59 @@
...
+        COMSHARE.delete_share_by_api('nfs', nfs_data['api_path'])
+        DATASET.delete_dataset_by_api(nfs_data['api_path'])
+
+    @allure.story("Create New NFS Share")
+    def test_create_new_nfs_share(self, nfs_data) -> None:
+        """
+        This test creates a new NFS share with the given data.
+        """
+        # Navigate to Shares page
+        NAV.navigate_to_shares()
```
Comment (markbainix) [inline test_cases/webui/shares/nfs/test_create_new_nfs_share.py:41]: suggest moving into `setup_test()`

File: test_cases/webui/shares/nfs/test_delete_new_nfs_share.py [hunk 5]
```diff
@@ -6,32 +7,50 @@
...
+        # Verify share detached from dataset
+        NAV.navigate_to_datasets()
+        DATASET.expand_dataset('tank')
+        DATASET.select_dataset(nfs_data['dataset_name'])
+        assert not DATASET.assert_dataset_share_attached(nfs_data['dataset_name'], 'nfs')
+        assert not DATASET.assert_dataset_roles_share_icon(nfs_data['dataset_name'], 'nfs')
+
+        # Verify share deleted from Shares page
+        NAV.navigate_to_shares()
+        assert not COMSHARE.is_share_visible('nfs', nfs_data['share_page_path'])
```
Comment (markbainix) [inline test_cases/webui/shares/nfs/test_delete_new_nfs_share.py:56]: change `not` to `is False`

File: test_cases/webui/shares/nfs/test_create_new_nfs_share.py [hunk 6]
```diff
@@ -6,39 +7,59 @@
...
+        # Verify share attachment to dataset
+        NAV.navigate_to_datasets()
+        DATASET.expand_dataset('tank')
+        DATASET.select_dataset(nfs_data['dataset_name'])
+        assert DATASET.assert_dataset_share_attached(nfs_data['dataset_name'], 'nfs')
+        assert DATASET.assert_dataset_roles_share_icon(nfs_data['dataset_name'], 'nfs')
+
+        # Verify share displayed on shares page
+        NAV.navigate_to_shares()
+        assert COMSHARE.assert_share_path('nfs', nfs_data['share_page_path'])
```
Comment (markbainix) [inline test_cases/webui/shares/nfs/test_create_new_nfs_share.py:63]: assert should have `is True`

File: test_cases/webui/shares/nfs/test_edit_nfs_share.py [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+        # Navigate to the Shares page
+        NAV.navigate_to_shares()
+        assert COMSHARE.assert_share_card_displays('nfs')
+
+        # Edit the NFS share with advanced options
+        assert COMSHARE.assert_share_path('nfs', nfs_data['share_page_path'])
+        COMSHARE.click_edit_share('nfs', nfs_data['share_page_path'])
+        COM.set_checkbox('enabled')
+        NFS.click_add_networks_button()
+        WebUI.delay(1)
```
Comment (markbainix) [inline test_cases/webui/shares/nfs/test_edit_nfs_share.py:58]: if these delays are needed, suggest moving them into the methods and not have them in the test case

File: test_cases/webui/shares/nfs/test_nfs_share_card_ui.py [hunk 8]
```diff
@@ -8,74 +9,87 @@
...
+        assert COMSHARE.assert_share_path('nfs', "/mnt/tank/sharefour")
+        assert not COMSHARE.assert_share_path('nfs', "/mnt/tank/sharefive")
 
-    # Verify NFS View All page UI
-    assert COMSHARE.assert_share_view_all_page_add_button('nfs')
-    assert COMSHARE.assert_share_view_all_page_enabled_button('nfs', "/mnt/tank/shareone")
-    assert COMSHARE.assert_share_view_all_page_button_by_name('nfs', "/mnt/tank/shareone", 'edit')
-    assert COMSHARE.assert_share_view_all_page_button_by_name('nfs', "/mnt/tank/shareone", 'delete')
+        # Verify all shares display after clicking the View All button
+        COM.click_link(f'nfs-share-view-all')
```
Comment (markbainix) [inline test_cases/webui/shares/nfs/test_nfs_share_card_ui.py:83]: parameter doesn't need 'f' for formatting as there is no variable

**Review by markbainix — CHANGES_REQUESTED** [review]

File: test_cases/webui/shares/nfs/conftest.py (hunk 1 above)
Comment (jordan-mcgillivray005) [reply test_cases/webui/shares/nfs/conftest.py:16]: Agreed. Needless duplication of code

**Comment by ericbsd** [comment]
image

**Review by markbainix — APPROVED** [review]

**Pull request merged**